staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/list_width.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package request
+
+import (
+	"math"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+)
+
+const (
+	// the minimum number of seats a list request occupies
+	minimumSeatsList uint = 3
+
+	// the maximum number of seats a list request occupies
+	maximumSeatsList uint = 10
+)
+
+func newListWidthEstimator(countFn objectCountGetterFunc) WidthEstimatorFunc {
+	estimator := &listWidthEstimator{
+		countFn: countFn,
+	}
+	return estimator.estimate
+}
+
+type listWidthEstimator struct {
+	countFn objectCountGetterFunc
+}
+
+func (e *listWidthEstimator) estimate(r *http.Request) Width {
+	requestInfo, ok := apirequest.RequestInfoFrom(r.Context())
+	if !ok {
+		return Width{Seats: minimumSeatsList}
```

Comment:
Technically, no requestInfo should never happen, so just in case I prefer being on the safe side and returning maximumSeatsList.